blockchain/src/history/validity_store.rs

```diff
@@ -123,9 +140,14 @@ impl ValidityStore {
             "Pruning the validity store"
         );
 
-        if num_blocks > Policy::transaction_validity_window_blocks() {
+        // We need to keep at least 'validity_window_blocks' + 'blocks_per_batch' blocks
+        // to account for potential rebranches (when the window moves backwards).
```

Comment:
It might be worth mentioning this in the validity store documentation:
The validity store keeps at least 'validity_window_blocks' + 'blocks_per_batch' blocks